tests/test_lsst.py

```diff
@@ -0,0 +1,928 @@
+        self.assertNotEqual(self.wcs, wcs3)
+
+        wcs4 = LsstWCS(self.pointing, self.rotation, 'R:2,2 S:2,2')
+        self.assertNotEqual(self.wcs, wcs4)
+
+
+    def test_copy(self):
+        """
+        Test that copy() works
+        """
+        pointing = CelestialCoord(64.82*galsim.degrees, -16.73*galsim.degrees)
+        rotation = 116.8*galsim.degrees
+        chip_name = 'R:1,2 S:2,2'
+        wcs0 = LsstWCS(pointing, rotation, chip_name)
+        wcs0._newOrigin(galsim.PositionI(112, 4))
+        wcs1 = wcs0.copy()
+        self.assertEqual(wcs0, wcs1)
+
+        wcs0._newOrigin(galsim.PositionI(66, 77))
+        self.assertNotEqual(wcs0, wcs1)
+
+
+    def test_pickling(self):
+        """
+        Test that LsstWCS can be pickled and un-pickled
+        """
+        path, filename = os.path.split(__file__)
+        file_name = os.path.join(path,'scratch_space','pickle_LsstWCS.txt')
+
+        import pickle
+
+        with open(file_name, 'w') as output_file:
+             pickle.dump(self.wcs, output_file)
+
+        with open(file_name, 'r') as input_file:
+            wcs1 = pickle.load(input_file)
+
+        self.assertEqual(self.wcs, wcs1)
+
+        if os.path.exists(file_name):
+            os.unlink(file_name)
+
+
+    def test_passing_camera_by_hand(self):
+        """
+        Test that you can pass a camera from one WCS to another
+        """
+
+        with warnings.catch_warnings(record=True) as ww:
+            wcs1 = LsstWCS(self.pointing, self.rotation, chip_name='R:0,1 S:1,1', camera=self.wcs.camera)
+
+        self.assertEqual(len(ww), 0)
+
+
+        # verify that, if the camera does not have the pointing or rotation angle you want,
+        # a new camera will be instantiated
+        with warnings.catch_warnings(record=True) as ww:
+            wcs1 = LsstWCS(galsim.CelestialCoord(0.0*galsim.degrees, 0.0*galsim.degrees), self.rotation,
+                           chip_name='R:0,1 S:1,1', camera=self.wcs.camera)
+
+        self.assertEqual(str(ww[0].message),
+                         "The camera you passed to LsstWCS does not have the same\n"
+                         "pointing and rotation angle as you asked for for this WCS.\n"
+                         "LsstWCS is creating a new camera with the pointing and\n"
+                         "rotation angle you specified in the constructor for LsstWCS.")
+
+
+        with warnings.catch_warnings(record=True) as ww:
+            wcs1 = LsstWCS(self.pointing, 49.0*galsim.degrees,
+                           chip_name='R:0,1 S:1,1', camera=self.wcs.camera)
+
+        self.assertEqual(str(ww[0].message),
+                         "The camera you passed to LsstWCS does not have the same\n"
+                         "pointing and rotation angle as you asked for for this WCS.\n"
+                         "LsstWCS is creating a new camera with the pointing and\n"
+                         "rotation angle you specified in the constructor for LsstWCS.")
```

Comment:
We were running afoul of python's natural tendency to filter warnings such that the same warning is not raised more than once (the test in question tries to provoke the warning in two different ways).  I have added a warning filter to allow the warning to be raised multiple times (since, presumably, we want the user to be alerted every time they try to create a WCS with a camera that is not configured to the same world origin as the WCS).